### PR TITLE
refactor: eliminate double re-execution in AsyncChecker

### PR DESCRIPTION
**Pull request by omerfirmak**
## 1. Purpose or design rationale of this PR

as pointed out by @lispc as well, `core.ApplyTransaction` is not really needed. This will also help with reducing contention on the state database, that can potentially interfere with block building in worker.

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [ ] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [ ] This PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Review by 0xmountaintop — DISMISSED** [review]

**Review by colinlyguo — DISMISSED** [review]

File: rollup/ccc/async_checker.go [hunk 1]
```diff
@@ -208,39 +206,14 @@ func (c *AsyncChecker) checkerTask(block *types.Block, ccc *Checker, forkCtx con
...
-
-	// 1. we have to check circuit capacity before `core.ApplyTransaction`,
-	// because if the tx can be successfully executed but circuit capacity overflows, it will be inconvenient to revert.
-	// 2. even if we don't commit to the state during the tracing (which means `clearJournalAndRefund` is not called during the tracing),
-	// the `refund` value will still be correct, because:
-	// 2.1 when starting handling the first tx, `state.refund` is 0 by default,
-	// 2.2 after tracing, the state is either committed in `core.ApplyTransaction`, or reverted, so the `state.refund` can be cleared,
-	// 2.3 when starting handling the following txs, `state.refund` comes as 0
+func (c *AsyncChecker) checkTx(parent *types.Block, header *types.Header, state *state.StateDB, tx *types.Transaction, ccc *Checker) (*types.RowConsumption, error) {
 	trace, err := tracing.NewTracerWrapper().CreateTraceEnvAndGetBlockTrace(c.bc.Config(), c.bc, c.bc.Engine(), c.bc.Database(),
```
Comment (colinlyguo) [inline rollup/ccc/async_checker.go:210]: Why is `commitStateAfterApply` true after the change?

Comment (omerfirmak) [reply rollup/ccc/async_checker.go:210]: It is to more closely mimic `core.ApplyTransaction` behaviour.
see `applyTransaction`

Comment (omerfirmak) [reply rollup/ccc/async_checker.go:210]: But I believe it is unnecessary, will set it to false.

Comment (colinlyguo) [reply rollup/ccc/async_checker.go:210]: got it. the original reason for this comment is that I'm afraid that setting the flag as "true" will modify the state db (maybe wrong, if it's wrong pls point it out).

Comment (omerfirmak) [reply rollup/ccc/async_checker.go:210]: > But I believe it is unnecessary, will set it to false.

Looks like it was necessary :D 


```
thread '<unnamed>' panicked at /root/.cargo/git/checkouts/zkevm-circuits-462c2cba34f9b301/1f2f15c/bus-mapping/src/evm/opcodes/sstore.rs:173:9:
assertion `left == right` failed: expected refund 0 is not equal to current 4800
  left: 4800
 right: 0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
ERROR[09-13|06:49:49.921] fail to apply_tx in Checker              id=6                err="estimate_circuit_capacity (id: 6) error in apply_tx, error: Any { .. }"
thread '<unnamed>' panicked at /root/.cargo/git/checkouts/zkevm-circuits-462c2cba34f9b301/1f2f15c/bus-mapping/src/evm/opcodes/sstore.rs:173:9:
assertion `left == right` failed: expected refund 0 is not equal to current 72100
  left: 72100
 right: 0
ERROR[09-13|06:49:49.952] fail to apply_tx in Checker              id=4                err="estimate_circuit_capacity (id: 4) error in apply_tx, error: Any { .. }"
thread '<unnamed>' panicked at /root/.cargo/git/checkouts/zkevm-circuits-462c2cba34f9b301/1f2f15c/bus-mapping/src/evm/opcodes/sstore.rs:173:9:
assertion `left == right` failed: expected refund 0 is not equal to current 39800
  left: 39800
 right: 0
ERROR[09-13|06:49:49.996] fail to apply_tx in Checker              id=5                err="estimate_circuit_capacity (id: 5) error in apply_tx, error: Any { .. }"
thread '<unnamed>' panicked at /root/.cargo/git/checkouts/zkevm-circuits-462c2cba34f9b301/1f2f15c/bus-mapping/src/evm/opcodes/sstore.rs:173:9:
assertion `left == right` failed: expected refund 0 is not equal to current 22700
  left: 22700
 right: 0
ERROR[09-13|06:49:50.059] fail to apply_tx in Checker              id=0                err="estimate_circuit_capacity (id: 0) error in apply_tx, error: Any { .. }"
```

> will modify the state db

do you mean the permanent storage? Unless we call `WriteBlockWithState`like we do in worker, nothing will be written to the DB.

Comment (colinlyguo) [reply rollup/ccc/async_checker.go:210]: thanks, Omer, previously I misunderstood this, this PR will do `state.Finalise` in GetBlockTrace. resolved this comment.

**Review by colinlyguo — APPROVED** [review]

**Review by 0xmountaintop — APPROVED** [review]

**Pull request merged**